filesystems/s3/src/main/java/org/lerch/s3fs/S3FileSystemProvider.java

```diff
@@ -91,49 +90,19 @@ public String getScheme() {
 
     @Override
     public FileSystem newFileSystem(URI uri, Map<String, ?> env) {
-        return newFileSystem(uri, env, props -> createFileSystem(uri, props), true);
-    }
-
-    /**
-     * Get existing filesystem based on a combination of URI and env settings. Create new filesystem otherwise.
-     *
-     * @param uri URI of existing, or to be created filesystem.
-     * @param env environment settings.
-     * @return new or existing filesystem.
-     */
-    public FileSystem getFileSystem(URI uri, Map<String, ?> env, S3Client client) {
-        return newFileSystem(uri, env, props -> createFileSystem(uri, props, client), false);
-    }
-
-    private FileSystem newFileSystem(URI uri, Map<String, ?> env,
-                                     Function<Properties, S3FileSystem> createFileSystemFunc,
-                                     boolean throwExceptionIfAlreadyExists) {
         validateUri(uri);
         // get properties for the env or properties or system
         Properties props = getProperties(uri, env);
         validateProperties(props);
         // try to get the filesystem by the key
         String key = getFileSystemKey(uri, props);
-        FileSystemAlreadyExistsException alreadyExistsException = new FileSystemAlreadyExistsException("File system " + uri.getScheme() + ':' + key + " already exists");
-        if (!fileSystems.containsKey(key)) {
-            synchronized (fileSystems) {
```

Comment:
Double-check locking was introduced here for a reason: we had race condition happening in our tests https://broadworkbench.atlassian.net/browse/BA-6156

Did you address that elsewhere?

Reply:
I will revert this

Reply:
Looks like it still remains deleted?

Reply:
I missed that. Adding it back